src/sentry/integrations/discord/integration.py

```diff
@@ -238,39 +237,27 @@ def _get_discord_user_id(self, auth_code: str, url: str) -> str:
         integration.
 
         """
-        form_data = f"client_id={self.application_id}&client_secret={self.client_secret}&grant_type=authorization_code&code={auth_code}&redirect_uri={url}"
-        headers = {
-            "Content-Type": "application/x-www-form-urlencoded",
-        }
-
         try:
-            response = self.client.post(
-                "https://discord.com/api/v10/oauth2/token",
-                json=False,
-                data=form_data,
-                headers=headers,
-            )
-            token = response["access_token"]  # type: ignore
+            access_token = None
+            try:
+                access_token = self.client.get_access_token(auth_code, url)
+            except ApiError as e:
+                logger.exception(
+                    "discord.install.failed_to_get_access_token", extra={"error": str(e)}
+                )
+                raise IntegrationError("Failed to get Discord access token.") from e
+
+            try:
+                user_id = self.client.get_user_id(access_token)
+                return user_id
```

Comment:
I'd put this below the except. Also why change it from `raise ...` to `raise ... from e`?

Reply:
Whoops! Didn't mean to add the `from e`.